packages/addons-inject/index.js

```diff
@@ -0,0 +1,309 @@
+
+  // Log some debugging data
+  console.log(`ContentType: ${contentType}`);
+  console.log(`X-RTD-Force-Addons: ${forceAddons}`);
+  console.log(`X-RTD-Hosting-Integrations: ${injectHostingIntegrations}`);
+  console.log(`HTTP status: ${httpStatus}`);
+
+  // Debug mode for some test cases. This is just for triggering an exception
+  // from tests. There might be a way to conditionally enable this, but I had
+  // trouble getting Wrangler vars to work at all.
+  const throwError = headers.get("X-RTD-Throw-Error");
+  if (throwError) {
+    console.log(`Throw error: ${throwError}`);
+  }
+
+  // Get project/version slug from headers inject by El Proxito
+  const projectSlug = headers.get("X-RTD-Project") || "";
+  const versionSlug = headers.get("X-RTD-Version") || "";
+  const resolverFilename = headers.get("X-RTD-Resolver-Filename") || "";
+
+  // Check to decide whether or not inject Addons library. We only do this for
+  // `text/html` content types.
+  if (contentType.includes("text/html")) {
+    // Remove old implementation of our flyout and inject the new addons if the following conditions are met:
+    //
+    // - header `X-RTD-Force-Addons` is present (user opted-in into new beta addons)
+    // - header `X-RTD-Hosting-Integrations` is not present (added automatically when using `build.commands`)
+    //
+    if (forceAddons === "true" && injectHostingIntegrations === "false") {
+      let rewriter = new HTMLRewriter();
+
+      // Remove by selector lookup
+      for (const script of AddonsConstants.removalScripts) {
+        rewriter.on(script, new removeElement());
+      }
+      for (const link of AddonsConstants.removalLinks) {
+        rewriter.on(link, new removeElement());
+      }
+      for (const element of AddonsConstants.removalElements) {
+        rewriter.on(element, new removeElement());
+      }
+
+      // TODO match the pattern above and make this work
+      // NOTE: I wasn't able to reliably remove the "<script>" that parses
+      // the "READTHEDOCS_DATA" defined previously, so we are keeping it for now.
+      //
+      // rewriter.on(readthedocsDataParse, new removeElement())
+      // rewriter.on(readthedocsData, new removeElement())
+
+      return rewriter
+        .on("head", new addPreloads())
+        .on(
+          "head",
+          new addMetaTags(
+            projectSlug,
+            versionSlug,
+            resolverFilename,
+            httpStatus,
+          ),
+        )
+        .on("*", {
+          // This is only used for testing purposes. This is used to mimic an
+          // error being thrown during the request to origin. There are cases
+          // that are difficult to model here, like a large page size timing out
+          // the request.
+          element(element) {
+            if (throwError) {
+              throw new Error("Manually triggered error in transform");
+            }
+          },
+        })
+        .transform(await response.clone());
```

Comment:
Do we need to clone the response? Looks like the previous code, we didn't have to.